### PR TITLE
RHOAI: Build VSCode notebook on RHEL9 base

### PR DESCRIPTION
**Pull request by atheo89**
Ref. Issue: https://github.com/opendatahub-io/notebooks/issues/283

Depend on: https://github.com/openshift/release/pull/45089

This PR incorporates the VSCode notebook to RHOAI.

**How has been tested:**
The behavior of the notebook looks the same as the one based on C9S in ODH.
The RHEL base image that used is : https://catalog.redhat.com/software/containers/rhel9/python-39/61a6101fbfd4a5234d59629d?architecture=amd64&image=65301f5a53a51a886582fba1

**1. On a Local Run** 
1. Build the notebook: `make codeserver-rhel9-python-3.9`
2. Run the VSCode locally: 
```podman run --network=host --name validation-container quay.io/rh_ee_atheodor/workbench-images:codeserver-rhel9-python-3.9-2023b_20231106```
  
image

Once you finish the inspection: `podman stop validation-container && podman rm validation-container`

**2. On the Cluster** 

Deploy on the cluster: `make deploy-rhel9-codeserver-rhel9-python-3.9`
Run the Test: `make validate-codeserver-image image=codeserver-rhel9-python-3.9-2023b_20231106` ensure that the test completed successfully
Undeploy: `make undeploy-rhel9-codeserver-rhel9-python-3.9`

### CODE REVIEW

File: codeserver/rhel9-python-3.9/Dockerfile
Comment (jstourac) [inline codeserver/rhel9-python-3.9/Dockerfile:9]:
```suggestion
      io.k8s.display-name="Code Server (VS Code) image with python 3.9 based on Red Hat Enterprise Linux 9" \
```

File: codeserver/rhel9-python-3.9/Dockerfile
Comment (jstourac) [inline codeserver/rhel9-python-3.9/Dockerfile:10]:
```suggestion
      io.k8s.description="Code Server (VS Code) image with python 3.9 based on Red Hat Enterprise Linux 9" \
```

File: codeserver/rhel9-python-3.9/Dockerfile
Comment (jstourac) [inline codeserver/rhel9-python-3.9/Dockerfile:7]:
```suggestion
      summary="Code Server (VS Code) image with python 3.9 based on Red Hat Enterprise Linux 9" \
```

File: codeserver/rhel9-python-3.9/Dockerfile
Comment (jstourac) [inline codeserver/rhel9-python-3.9/Dockerfile:6]:
```suggestion
LABEL name="odh-notebook-code-server-rhel9-python-3.9" \
```

File: codeserver/rhel9-python-3.9/Dockerfile
Comment (jstourac) [inline codeserver/rhel9-python-3.9/Dockerfile:8]:
```suggestion
      description="Code Server (VS Code) image with python 3.9 based on Red Hat Enterprise Linux 9" \
```

File: codeserver/rhel9-python-3.9/Dockerfile
Comment (jstourac) [inline codeserver/rhel9-python-3.9/Dockerfile:14]: Wrong links on these two lines.

Also, at the moment we point in downstream to the upstream locations... this is something that we should update in the future probably as a bulk update - eventually utilizing something from opendatahub-io/notebooks#309.

File: base/rhel9-python-3.9/Dockerfile
Comment (jstourac) [inline base/rhel9-python-3.9/Dockerfile:7]:
```suggestion
      io.k8s.description="Base Python 3.9 builder image based on Red Hat Enterprise Linux 9 for ODH notebooks" \
```

Just to unify names we use across these descriptions.

File: base/rhel9-python-3.9/Dockerfile
Comment (jstourac) [inline base/rhel9-python-3.9/Dockerfile:6]:
```suggestion
      io.k8s.display-name="Python 3.9 Red Hat Enterprise Linux 9 base image for ODH notebooks" \
```

File: manifests/base/params.env
Comment (jstourac) [inline manifests/base/params.env:19]: I suppose that this is something that is gonna be updated later, right?

**Review by jstourac — CHANGES_REQUESTED** [review]
Hi @atheo89, thank you for this. I've put a few comments here. Otherwise LGTM.

File: manifests/base/params.env
Comment (atheo89) [reply manifests/base/params.env:19]: Exactly, i will open a follow up PR to fill this value. Once this PR get merged the CI will build the image and eventually I will be able to use it on the params.env

**Comment by atheo89** [comment]
/hold 

Checking the case if we can build the VSCode based on UBI9 on upstream and downstream,in favor of less maintenance and help QE to test it earlier on upstream first.

**Comment by atheo89** [comment]
Seems the rebase to UBI9 it works well.
This is the PR on upstream -> https://github.com/opendatahub-io/notebooks/pull/319

**Comment by atheo89** [comment]
Closing this since we were able to build code server notebook based on UBI9 base image, so no need to keep this open.
This PR https://github.com/red-hat-data-services/notebooks/pull/105 is going to introduce code server on downstream

**Pull request closed without merging**
